### DISCUSSION TITLE
react，chrome浏览器能显示轮播图的多张图片，但仍然提示说`Swiper` needs at least one child

### DISCUSSION THREAD

**Discussion by ken428965** [ant-design/ant-design-mobile#5797] [Q&A] [answered]
**已写的部分代码：**

```
function HouseDetail() {
const [houseinfo, setHouseinfo] = useState({
      houseImg: []
      }
  })
  function renderSwipers() {
    const { houseImg } = houseinfo
    console.log(houseImg)
    return houseImg.map(item => {
      const val = /^[\s\S]*(.jpg)$/
      if(val.test(item)==false) {
        return
      }
      return (
        <Swiper.Item key={item}>
          <a href="https://www.baidu.com">
            <img src={BASE_URL + item} alt="" />
          </a>
        </Swiper.Item>
      )
    })
  }
    return (
        {/* 轮播图 */}
        <div className={styles.slides}>
          {!isloading ? (
            <Swiper indicatorProps={{
            color: 'white',
        }} defaultIndex={0} loop autoplay infinite autoplayInterval={5000}>
              {renderSwipers()}
            </Swiper>
          ) : (
            ''
          )}
        </div>
}
```

**能console出图片数据，能显示多张轮播图，swiper和swiper.item里面已经有子元素，但浏览器就是一直提示说`Swiper` needs at least one child，请问为什么？**

**Comment by miracles1919** [comment, marked as answer]
代码能格式化一下么 ...

**Reply by ken428965** [reply]
您好。感谢回复~我在填写本讨论内容时是显示了格式化后的代码，但不知为何，发出后看到的是没有格式化的样子。请问哪个按钮可以格式化文本？

**Reply by afc163** [reply]
https://docs.github.com/cn/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code

**Reply by ken428965** [reply]
打开链接后看到里面都是英文。。。试过是应用翻译插件，但翻译不成功。。。请问您方便把本讨论里的代码复制到您的vscode里查看吗，这样或许能代码可读性的问题~

**Reply by MuxinFeng** [reply]
我猜是应该用 "```" 而不是 "`"

**Reply by ken428965** [reply]
感谢指教~~~已把代码改成具备可读性的样子了，请过目

**Reply by MuxinFeng** [reply]
emmm，其实可读性还是比较差，然后我把代码摘到编辑器里，还是解析不出来，要不你把括号、缩进再补全一下吧

**Reply by ken428965** [reply]
把括号和缩进补全，并把代码省略符// ...删除后，现在好点了吗？

**Reply by MuxinFeng** [reply]
就从目前的代码来看，你返回的是空数组，就算解析对了也会报无子元素的错。

**Reply by ken428965** [reply]
抱歉，虚心请教一下：看到那些轮播图片可以正常显示出来，证明里面有数据，然而返回的是空数组（就是没有数据），这个点我确实不太明白。
请问如何修改才能解决此故障？

**Reply by MuxinFeng** [reply]
https://codesandbox.io/s/hidden-dream-fg6ktc?file=/src/App.tsx

**Reply by MuxinFeng** [reply]
renderSwipers() 返回的数组里至少得有一个值，你渲染的不是 houseImg，所以它是否有值没有意义，因为经过test，可能就是空数组了

**Reply by ken428965** [reply]
原来如此~~~渲染的不是 houseImg，所以它是否有值没有意义，加上可能经过test就变成空数组了~~~
看到您分享的修改内容，非常感谢~~~
然而，考虑到我这边需要展示的图片有1000多张，如果在src里写固定图片地址会难以做到，因此把内容改成以下：
```
const { houseImg } = houseinfo;
    let arr = [];
    houseImg.map((item, index) => {
        let imgs = {
            imgSrc: `${BASE_URL}${item}`,
            id: index + 1
        };
        arr.push(imgs)
    })
    return arr.map(item => {
        return (
            <Swiper.Item key={item.id}>
              <a href="https://www.baidu.com">
                <img src={item.imgSrc} alt="" />
              </a>
            </Swiper.Item>
        )
    })

```

浏览器依然报那个需要有子元素的提示。。。不知道问题出在哪里。。。

**Reply by ken428965** [reply]
imgSrc里的地址是对的，因为可以正常显示多张轮播图片

**Reply by MuxinFeng** [reply]
把你要渲染的元素打印出来，如果不是空的也展示不了，就试试 {...renderItem()}

**Reply by ken428965** [reply]
问题解决了，原来在这里加两个并且即可：
```
{!isloading && houseImg && houseImg.length > 0 ? (
              <Swiper indicatorProps={{
                  color: 'white',
              }} defaultIndex={0} loop autoplay autoplayInterval={5000}>
                  {renderSwipers()}
              </Swiper>
          ) : (
            ''
          )}

```

感谢您的提醒与分享~~~

**Reply by ken428965** [reply]
问题解决了，原来在这里加两个并且即可：
```
{!isloading && houseImg && houseImg.length > 0 ? (
              <Swiper indicatorProps={{
                  color: 'white',
              }} defaultIndex={0} loop autoplay autoplayInterval={5000}>
                  {renderSwipers()}
              </Swiper>
          ) : (
            ''
          )}

```

感谢您的提醒与分享~~~

**Comment by MuxinFeng** [comment]
```
git status
git add
git commit
```
一例
<img width="895" alt="image" src="https://user-images.githubusercontent.com/46584765/201011965-fd7f7e19-1394-4b6c-9698-7033fc686c23.png">

**Comment by ken428965** [comment]
> **已写的部分代码：** ```// ... function HouseDetail() { const [houseinfo, setHouseinfo] = useState({ houseImg: [], // ... } }) // ... function renderSwipers() { const { houseImg } = houseinfo console.log(houseImg) return houseImg.map(item => { const val = /^[\s\S]*(.jpg)$/ if(val.test(item)==false) { return } return ( <Swiper.Item key={item}> <a href="https://www.baidu.com"> <img src={BASE_URL + item} alt="" /> </a> </Swiper.Item> ) }) } // ... return ( // ... {/* 轮播图 */} <div className={styles.slides}> {!isloading ? ( <Swiper indicatorProps={{ color: 'white', }} defaultIndex={0} loop autoplay infinite autoplayInterval={5000}> {renderSwipers()} </Swiper> ) : ( '' )} </div> // ... } // ...``` **能console出图片数据，能显示多张轮播图，swiper和swiper.item里面已经有子元素，但浏览器就是一直提示说`Swiper` needs at least one child，请问为什么？**
